### PR TITLE
Update QA tools

### PR DESCRIPTION
**Pull request by michalbundyra**
- Updated travis configuration to run tests with lowest/locked/latest dependencies.
- Added `composer.lock` to the repository.
- Added coveralls configuration and badge.
- Updated PHPDocs.
- Updated PHPUnit to version ^5.7.
- Changed link to the documentation (domain docs.zendframework.com instead of github.io).

__Please enable coveralls for the repository!__

### CODE REVIEW

**Comment by Ocramius** [comment]
@webimpress changed base branch, rebase needed. Please remember that for other PRs too :-)

**Comment by michalbundyra** [comment]
@Ocramius rebased, thanks! Will do the same for others then.

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -41,15 +41,15 @@ To run tests:
...
+  $ git clone git://github.com/zendframework/zend-expressive-fastroute.git
   $ cd zend-expressive-fastroute
   ```
 
 - Install dependencies via composer:
 
   ```console
   $ curl -sS https://getcomposer.org/installer | php --
-  $ ./composer.phar install
+  $ composer install
```
Comment (schnittstabil) [inline CONTRIBUTING.md:52]: This doesn't work. `curl … | php --` just downloads `composer.phar`, but does not install composer globally – Same issue as the one already mentioned at https://github.com/zendframework/zend-stratigility/pull/97#commitcomment-20593201 :unamused:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,15 +41,15 @@ To run tests:
 - Clone the repository:
 
   ```console
-  $ git clone git@github.com:zendframework/zend-expressive-fastroute.git
+  $ git clone git://github.com/zendframework/zend-expressive-fastroute.git
```
Comment (schnittstabil) [inline CONTRIBUTING.md:44]: I'm not sure about this. I've already mentioned my concerns at https://github.com/zendframework/zend-stratigility/pull/89#discussion_r97573832

**Pull request merged**